engine/src/main/java/org/terasology/rendering/particles/ParticleData.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2015 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.particles;
+
+import org.terasology.math.geom.Vector2f;
+import org.terasology.math.geom.Vector3f;
+import org.terasology.math.geom.Vector4f;
+
+/**
+ * Data object to store the data of a single particle.
+ * Used internally for swapping operations.
+ */
+public final class ParticleData {
+    // scalars
+    public float energy;
+
+    // 2d vectors
+    public final Vector2f textureOffset = new Vector2f();
+
+    // 3d vectors
+    public final Vector3f position = new Vector3f();
+    public final Vector3f previousPosition = new Vector3f();
+    public final Vector3f velocity = new Vector3f();
+    public final Vector3f scale = new Vector3f();
+
+    // 4d vectors
+    public final Vector4f color = new Vector4f();
```

Comment:
Hmmm... the internal representation of the Color class is an int. That's 4 bytes vs the 4x4=16 bytes of color precision of a Vector3f. From a performance, storage and bandwidth perspective of course it has its appeal, but I'd maybe offer it as a possible alternative rather than the default, perhaps an "HighPerformanceParticleData" class. 

I'd recommend leaving this as it is for the time being but writing a proposal issue about it, so we don't loose the idea.